app/api/Datasets.scala

```diff
@@ -1134,22 +1128,56 @@ class  Datasets @Inject()(
   private def getFilesWithinFolders(id: UUID, serveradmin: Boolean=false, max: Int = -1): List[JsValue] = {
     val output = new ListBuffer[JsValue]()
     var resultCount = 0
+    var current_folder : Option[Folder] = None
     datasets.get(id) match {
       case Some(dataset) => {
-        folders.findByParentDatasetId(id).map { folder =>
+        folders.findByParentDatasetId(id).foreach { folder =>
+          current_folder = Some(folder)
           files.get(folder.files).found.foreach(file => {
             if (max < 0 || resultCount < max) {
-              output += jsonFile(file, serveradmin)
+              output += jsonFileWithFolder(file, current_folder.get, serveradmin)
               resultCount += 1
             }
           })
+          print("done with folder")
         }
       }
       case None => Logger.error(s"Error getting dataset $id")
     }
     output.toList
   }
 
+  def jsonFileWithFolder(file: models.File, folder: models.Folder, serverAdmin: Boolean = false) : JsValue = {
```

Comment:
since this is pretty much a copy/paste from below, why not add tit to below as option with None as default?

Reply:
done